hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,302 @@
+					TerminationGracePeriodSeconds: &[]int64{90}[0],
+					Containers: []corev1.Container{
+						{
+							Name:            containerName,
+							Image:           ccm.image,
+							ImagePullPolicy: "Always",
+							Env: []corev1.EnvVar{
+								{
+									Name: "POD_IP_ADDRESS",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											FieldPath: "status.podIP",
+										},
+									},
+								},
+								{
+									Name:  "VPCCTL_CLOUD_CONFIG",
+									Value: fmt.Sprintf("%s/%s", configMapMountPath, configMapName),
+								},
+								{
+									Name:  "VPCCTL_PUBLIC_ENDPOINT",
+									Value: "true",
+								},
+							},
+							Command: []string{
+								"/bin/bash",
+								"-c",
+								strings.Join(commandToExec, " "),
+							},
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.IntOrString{IntVal: 10258},
+										Scheme: "HTTPS",
+									},
+								},
+								InitialDelaySeconds: 300,
+								TimeoutSeconds:      160,
+							},
+							Ports: []corev1.ContainerPort{
+								{
+									Name:          "https",
+									Protocol:      "TCP",
+									ContainerPort: 10258,
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									"cpu":    resource.MustParse("75m"),
+									"memory": resource.MustParse("60Mi"),
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      ccm.kubeConfig.Name,
+									MountPath: kubeConfigFileBasePath,
+								},
+								{
+									Name:      configMapName,
+									MountPath: configMapMountPath,
+								},
+								{
+									Name:      ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name,
+									MountPath: secretMountPath,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: ccm.kubeConfig.Name,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName:  ccm.kubeConfig.Name,
+									DefaultMode: &[]int32{400}[0],
+								},
+							},
+						},
+						{
+							Name: configMapName,
+							VolumeSource: corev1.VolumeSource{
+								ConfigMap: &corev1.ConfigMapVolumeSource{
+									DefaultMode:          &[]int32{420}[0],
+									LocalObjectReference: corev1.LocalObjectReference{Name: configMapName},
+								},
+							},
+						},
+						{
+							Name: ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name,
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName:  ccm.hcluster.Spec.Platform.PowerVS.ClusterControlManagerCreds.Name,
+									DefaultMode: &[]int32{400}[0],
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err = ccm.createOrUpdate(ccm.ctx, ccm.c, deployment, func() error {
```

Comment:
Same as the other,  the content of the deployment need to be set in here or this does nothing

Reply:
Ptal at the new code